src/psyclone/domain/lfric/lfric_loop.py

```diff
@@ -160,23 +151,23 @@ def lower_to_language_level(self):
                 self.loop_body.symbol_table.shallow_copy()
             loop.children[3] = self.loop_body.copy()
             self.replace_with(loop)
+            lowered_node = loop
         else:
-            # If loop_type is "null" we no need for a loop at all. So we set
-            # the bounds to 1 Remove the lower the children
-            dummy = self.scope.symbol_table.new_symbol(
-                "dummy", symbol_type=DataSymbol, datatype=INTEGER_TYPE)
-            self._variable = dummy
+            # If loop_type is "null" we not need a loop at all, just the kernel
+            # in its loop_body
             for child in self.loop_body.children:
                 child.lower_to_language_level()
-            one = Literal("1", INTEGER_TYPE)
-            loop = Loop.create(dummy, one.copy(), one.copy(), one.copy(), [])
-            loop.loop_body._symbol_table = \
-                self.loop_body.symbol_table.shallow_copy()
-            loop.children[3] = self.loop_body.copy()
-            self.replace_with(loop)
-
+            # TODO #1010: This restriction can be removed when also lowering the
+            # parent InvokeSchedule
+            if len(self.loop_body.children) > 1:
+                raise GenerationError(
```

Comment:
Strictly this should be a `NotImplementedError` unless that causes problems elsewhere?